index.js

```diff
@@ -58,19 +57,25 @@ class MultiStream extends stream.Readable {
     this._forwarding = false
   }
 
-  destroy (err) {
-    if (this.destroyed) return
-    this.destroyed = true
+  _destroy (err, cb) {
+    cb = once(cb)
 
-    if (this._current && this._current.destroy) this._current.destroy()
-    if (typeof this._queue !== 'function') {
-      this._queue.forEach(stream => {
-        if (stream.destroy) stream.destroy()
+    let streams = []
+    if (this._current) streams.push(this._current)
+    if (typeof this._queue !== 'function') streams = streams.concat(this._queue)
+
+    if (streams.length === 0) {
+      cb(err)
+    } else {
+      let counter = streams.length
+      streams.forEach(stream => {
+        destroy(stream, err, err => {
+          if (err || --counter === 0) {
+            cb(err)
```

Comment:
this can call the cb more than once if multiple streams errors

Reply:
See `cb = once(cb)` at the top of the function

Reply:
👍 in either case it should not call the cb until all streams have called back tho

Reply:
good point

Reply:
fixed